From fc01dae9a8f41b25f9fcaba6be62cb3a0bbefe1c Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 15 Jan 2024 12:16:12 +0100
Subject: libstdc++: fix static assertion in NodeUuidEquality

libstdc++ equality checks in static assertion that it is possible to
compare for equality base::Uuid to BookmarkNode*. This was a missing
operator in NodeUuidEquality that this changeset adds.

Bug: 957519
Change-Id: Icc9809cb43d321f0b3e3394ef27ab55672aec5e7
---
 components/bookmarks/browser/uuid_index.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/components/bookmarks/browser/uuid_index.h b/components/bookmarks/browser/uuid_index.h
index 77cb1a1a54dd9..639d6fefcd831 100644
--- a/components/bookmarks/browser/uuid_index.h
+++ b/components/bookmarks/browser/uuid_index.h
@@ -23,6 +23,10 @@ class NodeUuidEquality {
   bool operator()(const BookmarkNode* n1, const base::Uuid& uuid2) const {
     return n1->uuid() == uuid2;
   }
+
+  bool operator()(const base::Uuid& uuid1, const BookmarkNode* n2) const {
+    return uuid1 == n2->uuid();
+  }
 };
 
 // Used to hash BookmarkNode instances by UUID.
-- 
2.40.1

